spec/index.adoc

```diff
@@ -2283,6 +2297,7 @@ Arg = Blob;
 CallerId = Principal;
 
 Timestamp = Nat;
+CanHistoryIndex = Nat;
```

Comment:
Question about naming: Saying “CanHistoryIndex” works, but is a bit round-about – the number makes sense even if we don’t record the history, and I’d lean towards something like `CanisterGeneration`. But not important, of course.

Reply:
I'm not sure if `CanisterGeneration` should include controller changes which `CanisterHistoryIndex` (or `CanHistoryIndex` for short) does.

Reply:
Fair point. Maybe ok then.

Reply:
How about `CanisterVersion`? I think it still makes sense for the version of a canister to change when controllers are changed.

Actually, now that I think about it, would it make sense to have some kind of semantic versioning, for example a canister can have a version `x.y`, where `x` signifies a "major" change where the canister is upgraded, installed, reinstalled, and `y` signifies a "minor" change like a controller change, settings change, etc?

Reply:
The issue is how we version the legacy canisters. Currently, their index would be 0 because they have 0 change entries in their history.

Reply:
We have this issue for all approaches, right? With a single history index, we'll also start at zero for existing canisters, I presume?

Reply:
> We have this issue for all approaches

Not quite. I don't view index 0 as the first entry in the history, but as the number of entries in the history snapshot we want to reference (i.e., index 0 means "before history is recorded", index 1 means the first entry in the history etc.).

Reply:
During today's canisterOS meeting, we decided to keep a single index, in particular, because semantic versioning would typically bump the minor for code fixes and major for API changes whereas here we would bump the minor for non-code changes and major for all code changes.